pkg/cryptutil/certificates.go

```diff
@@ -219,6 +219,17 @@ func GenerateSelfSignedCertificate(domain string, configure ...func(*x509.Certif
 	return &cert, nil
 }
 
+// EncodeCertificate encodes a TLS certificate into PEM compatible byte slices.
+func EncodeCertificate(cert *tls.Certificate) (pemCertificateBytes, pemKeyBytes []byte) {
```

Comment:
To be on the safe side, do we want to return an error from this function since it's possible `x509.MarshalPKCS8PrivateKey` could fail?